### PR TITLE
Avoid dereference of nullptr in METAnalyzer

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

This was uncovered by UBSAN.

#### PR validation:

Ran the same workflow which was reporting the problem in the IB and no report happened with this change.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by jfernan2** [comment]
+1

File: DQMOffline/JetMET/src/METAnalyzer.cc [hunk 1]
```diff
@@ -1504,14 +1504,17 @@ void METAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetu
   //met=&(tcmetcoll->front());
   //}
   if (isPFMet_) {
+    assert(!pfmetcoll->empty());
```
Comment (fabiocos) [inline DQMOffline/JetMET/src/METAnalyzer.cc]: @Dr15Jones @jfernan2 are we expecting this instruction to abort the job? Is this the behaviour that we want in this DQM analyzer? Or should we set a different type of protection?

Comment (Dr15Jones) [reply DQMOffline/JetMET/src/METAnalyzer.cc]: The lines following lead to undefined behavior. In the case of programming errors I use `assert`. In the case of runtime problems I'd use exceptions. At the cursory glance I gave the code, it looked like if you got here with an empty collection it was a logic problem with the algorithm. Hence why I chose `assert`.

**Comment by fabiocos** [comment]
+1

**Pull request merged**